File: 2º/PDOO/ProyectoJAVA/src/civitas/S_irCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author nikitastetskiy
 */
public class S_irCarcel extends Sorpresa{
    
    private Tablero tablero;
    
    S_irCarcel(Tablero tablero, String texto){
        super(texto);
        this.tablero = tablero;
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
    }
    
    @Override
    public String toString() {
       return super.toString();
    }
}

File: 2º/PDOO/ProyectoRUBY/lib/s_ir_carcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative"sorpresa.rb"
require_relative"jugador.rb"

module Civitas
  class S_irCarcel<Sorpresa
    def initialize(tablero)
      super("Vas a la carcel")
      @tablero=tablero
    end
    def aplicarAJugador(actual, todos)
      super.informe(actual,todos)
      todos[actual].encarcelar(@tabla.numCasillaCarcel)
    end
    def to_s
      aux=super
      return aux
    end

  end
end